cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/PreventBrokerScaleDownCheck.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.operator.cluster.model.KafkaCluster;
+import io.strimzi.operator.common.AdminClientProvider;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.operator.resource.SecretOperator;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ListTopicsOptions;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.TopicPartitionInfo;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.HashSet;
+/**
+ * Class which contains several utility function which check if broker scale down can be done or not.
+ */
+public class PreventBrokerScaleDownCheck {
+    /**
+     * Logger
+     */
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(PreventBrokerScaleDownCheck.class.getName());
+
+    /**
+     * Checks if broker contains any partition replicas when scaling down
+     *
+     * @param reconciliation       Reconciliation marker
+     * @param secretOperator       Secret operator for working with Secrets
+     * @param adminClientProvider  Used to create the Admin client instance
+     * @param idsToBeRemoved       Ids to be removed
+
+     * @return  returns a boolean future based on the outcome of the check
+     */
+    public Future<Set<Integer>> canScaleDownBrokers(Reconciliation reconciliation,
+                                                    Set<Integer> idsToBeRemoved, SecretOperator secretOperator, AdminClientProvider adminClientProvider) {
+
+        Promise<Set<Integer>> cannotScaleDown = Promise.promise();
+        ReconcilerUtils.clientSecrets(reconciliation, secretOperator)
+                .compose(compositeFuture -> {
+                    Promise<Void> resultPromise = Promise.promise();
+
+                    final Future<Collection<TopicDescription>> descriptions;
+                    try {
+                        String bootstrapHostname = KafkaResources.bootstrapServiceName(reconciliation.name()) + "." + reconciliation.namespace() + ".svc:" + KafkaCluster.REPLICATION_PORT;
+                        LOGGER.debugCr(reconciliation, "Creating AdminClient for cluster {}/{}", reconciliation.namespace());
```

Comment:
The debug command is looking for two parameters/placeholders but you are passing one, just the namespace.